addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/AbstractThingHandler.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neeo.handler;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.openhab.binding.neeo.internal.NeeoBrainApi;
+
+/**
+ * An abstract extension of {@link BaseThingHandler} that will add utility functions to all subclasses.
+ *
+ * @author Tim Roberts - initial contribution
+ */
+abstract class AbstractThingHandler extends BaseThingHandler {
+
+    /**
+     * Instantiates a new abstract thing handler from the given {@link Thing}
+     *
+     * @param thing the thing
+     */
+    public AbstractThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Returns the {@link NeeoBrainApi} associated with this {@link Bridge}. Will iterate through
+     * the parent handlers until one returns a {@link NeeoBrainApi} or null if not found
+     *
+     * @return the {@link NeeoBrainApi} or null if not found
+     */
+    @Nullable
+    public NeeoBrainApi getNeeoBrainApi() {
```

Comment:
Instead of having this method and the one below twice (copy&paste), how about moving it to your util class with a `Bridge bridge` parameter?